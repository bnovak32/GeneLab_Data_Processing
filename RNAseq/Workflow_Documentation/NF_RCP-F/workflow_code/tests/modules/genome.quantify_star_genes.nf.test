nextflow_process {

    name "Test Process QUANTIFY_STAR_GENES"
    script "modules/genome.nf"
    process "QUANTIFY_STAR_GENES"

    test("Baseline_ON_GLDS194:Should run without failures") {

        when {
            params {
                // define parameters here. Example:
                // outdir = "tests/results"
            }
            process {
                """
                // define inputs of the process here. Example:
                input[0] = file("test-datasets-extended/testdata/GLDS-194/mock/samples.txt")
                input[1] = file("test-datasets-extended/testdata/GLDS-194/02-STAR_Alignment/**/*_ReadsPerGene.out.tab")
                input[2] = 'antisense'
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out.publishables).match()
        }

    }

    test("Issue_60_Test_ON_Mock:Should run without failures") {

        when {
            params {
                // define parameters here. Example:
                // outdir = "tests/results"
            }
            process {
                """
                // define inputs of the process here. Example:
                input[0] = file("test-datasets-extended/testdata/mocks/overlapping_samplenames/samples.txt")
                input[1] = file("test-datasets-extended/testdata/mocks/overlapping_samplenames/*.out.tab")
                input[2] = 'antisense'
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out.publishables).match()
        }

    }

}
