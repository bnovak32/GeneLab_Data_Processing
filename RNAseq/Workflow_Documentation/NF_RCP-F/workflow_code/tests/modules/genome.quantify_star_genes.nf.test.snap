{
    "Baseline_ON_GLDS194:Should run without failures": {
        "content": [
            [
                [
                    "STAR_Unnormalized_Counts.csv:md5,ee11a0a026cad29f7d07982458749fdc",
                    "STAR_NumNonZeroGenes.csv:md5,614cfee77313d40f5722609bdf848aaf"
                ]
            ]
        ],
        "timestamp": "2023-01-25T20:38:18+0000"
    },
    "Issue_60_Test_ON_Mock:Should run without failures": {
        "content": [
            [
                [
                    "STAR_Unnormalized_Counts.csv:md5,699cfa8a1c81148197a2898b25bdcc4d",
                    "STAR_NumNonZeroGenes.csv:md5,447a4bc9208e065b011e53fc2557a30f"
                ]
            ]
        ],
        "timestamp": "2023-01-25T20:38:18+0000"
    }
}